### PR TITLE
Add and plumb through the ssh type.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

File: pkg/pki/ssh/ssh.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return []byte(Armor(s.signature, s.pk)), nil
+}
+
+// Verify implements the pki.Signature interface
+func (s Signature) Verify(r io.Reader, k interface{}) error {
+	if s.signature == nil {
+		return fmt.Errorf("ssh signature has not been initialized")
+	}
+
+	message, err := ioutil.ReadAll(r)
```
Comment (bobcallaway) [inline pkg/pki/ssh/ssh.go]: no need to do this, just pass through r to Verify

File: pkg/pki/ssh/ssh.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	pk        ssh.PublicKey
+}
+
+// NewSignature creates and Validates an ssh signature object
+func NewSignature(r io.Reader) (*Signature, error) {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	sig, pk, err := Decode(string(b))
```
Comment (bobcallaway) [inline pkg/pki/ssh/ssh.go]: nit: why do `Armor` and `Decode` return/take string arguments when it appears the first thing you do is convert back to `[]byte`?

Comment (bobcallaway) [reply pkg/pki/ssh/ssh.go]: also, should Decode just return the `Signature` struct as defined on line 28?

File: pkg/pki/ssh/ssh.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		return nil, err
+	}
+
+	return &PublicKey{key: key}, nil
+}
+
+// CanonicalValue implements the pki.PublicKey interface
+func (k PublicKey) CanonicalValue() ([]byte, error) {
+	if k.key == nil {
+		return nil, fmt.Errorf("x509 public key has not been initialized")
```
Comment (bobcallaway) [inline pkg/pki/ssh/ssh.go]: s/x509/ssh

File: pkg/pki/ssh/ssh.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	}
+	return &Signature{
+		signature: sig,
+		pk:        pk,
+	}, nil
+}
+
+// CanonicalValue implements the pki.Signature interface
+func (s Signature) CanonicalValue() ([]byte, error) {
+	return []byte(Armor(s.signature, s.pk)), nil
```
Comment (bobcallaway) [inline pkg/pki/ssh/ssh.go:47]: from above - change return type of Armor to []byte?

**Review by bobcallaway — APPROVED** [review]
lgtm, good work!

**Review by lukehinds — APPROVED** [review]
+1

**Pull request merged**